### PR TITLE
Fix iso provisioning

### PR DESCRIPTION
**Pull request by borod108**
Fix provisioning VM from ISO

### CODE REVIEW

**Comment by borod108** [comment]
@miq-bot add-label wip
@miq-bot add-label fine/yes

**Comment by borod108** [comment]
@jhernand can you please review? (especially regarding the detach_floppy)

File: app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb [hunk 1]
```diff
@@ -102,36 +102,47 @@ def powered_on_in_provider?(vm)
...
+        )
+      end
     rescue OvirtSDK4::Error
       raise OvirtServices::VmNotReadyToBoot
     end
 
+    def detach_floppy(operation)
+      operation.destination.with_provider_object(VERSION_HASH) do |vm_service|
+        vm = vm_service.get
+        vm.update(:payloads => [])
```
Comment (jhernand) [inline app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb]: The `update` method is a method of the service, not a method of the object. That means that this should be like this:

```ruby
vm_service.update(:payloads => [])
```

**Comment by oourfali** [comment]
Boris - is it still wip?
What tests are left to do?
Do we want Ilanit to test?

**Comment by borod108** [comment]
Yes it is, the tests are failing on travis but not on local. I have no clue
why yet.
Also I validated for v3 but not for v4, there was some small fix I did
there just now and I want to validate.
I want to test it myself, I want to take full responsibility for this and
make sure it works.
When I am done validating I will remove the wip and also ask Ilanit to give
me an env to patch with it so she can test.

**Comment by borod108** [comment]
@masayag please review

File: spec/models/manageiq/providers/redhat/infra_manager/provision_via_pxe/state_machine_spec.rb [hunk 2]
```diff
@@ -5,6 +5,7 @@
       template = FactoryGirl.create(:template_redhat, :ext_management_system => ems)
       @vm = FactoryGirl.create(:vm_redhat, :ext_management_system => ems)
       options = {:src_vm_id => template.id}
+
```
Comment (masayag) [inline spec/models/manageiq/providers/redhat/infra_manager/provision_via_pxe/state_machine_spec.rb]: why is this file part of this PR?

**Comment by borod108** [comment]
The relevant bz is: https://bugzilla.redhat.com/show_bug.cgi?id=1448137

**Comment by chrispy1** [comment]
@miq-bot add-label blocker

**Comment by simaishi** [comment]
Fine backport (to manageiq repo) details:

```
$ git log -1
commit 3b8002d0993d2fe98cc11897dc8357e5de198c6e
Author: Oved Ourfali <oourfali@redhat.com>
Date:   Mon May 8 20:50:07 2017 +0300

    Merge pull request #25 from borod108/bugs/iso_provisioning
    
    Fix iso provisioning
    (cherry picked from commit 402df6ed731140bb39e2d558f61a9f7ed6fb5278)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1449190
```

**Pull request merged**